plugins/modules/ec2_instance.py

```diff
@@ -1337,15 +1337,47 @@ def value_wrapper(v):
     ]
 
     for mapping in param_mappings:
-        if params.get(mapping.param_key) is not None and mapping.instance_key not in skip:
-            value = AWSRetry.jittered_backoff()(ec2.describe_instance_attribute)(Attribute=mapping.attribute_name, InstanceId=id_)
-            if params.get(mapping.param_key) is not None and value[mapping.instance_key]['Value'] != params.get(mapping.param_key):
-                arguments = dict(
-                    InstanceId=instance['InstanceId'],
-                    # Attribute=mapping.attribute_name,
-                )
-                arguments[mapping.instance_key] = mapping.add_value(params.get(mapping.param_key))
-                changes_to_apply.append(arguments)
+        if params.get(mapping.param_key) is None:
+            continue
+        if mapping.instance_key in skip:
+            continue
+
+        value = AWSRetry.jittered_backoff()(ec2.describe_instance_attribute)(Attribute=mapping.attribute_name, InstanceId=id_)
+        if value[mapping.instance_key]['Value'] != params.get(mapping.param_key):
+            arguments = dict(
+                InstanceId=instance['InstanceId'],
+                # Attribute=mapping.attribute_name,
+            )
+            arguments[mapping.instance_key] = mapping.add_value(params.get(mapping.param_key))
+            changes_to_apply.append(arguments)
+
+    if params.get('security_group') or params.get('security_groups'):
+        value = AWSRetry.jittered_backoff()(ec2.describe_instance_attribute)(Attribute="groupSet", InstanceId=id_)
+        # managing security groups
+        if params.get('vpc_subnet_id'):
+            subnet_id = params.get('vpc_subnet_id')
+        else:
+            default_vpc = get_default_vpc(ec2)
+            if default_vpc is None:
+                module.fail_json(
+                    msg="No default subnet could be found - you must include a VPC subnet ID (vpc_subnet_id parameter) to create an instance")
```

Comment:
This error should probably be about modifying security groups instead of creating an instance

Reply:
What do you think about the new verbiage?